mix.lock

```diff
@@ -1,13 +1,18 @@
-  "decimal": {:hex, :decimal, "1.4.0", "fac965ce71a46aab53d3a6ce45662806bdd708a4a95a65cde8a12eb0124a1333", [:mix], [], "hexpm"},
-  "earmark": {:hex, :earmark, "1.2.3", "206eb2e2ac1a794aa5256f3982de7a76bf4579ff91cb28d0e17ea2c9491e46a4", [:mix], [], "hexpm"},
-  "ex_doc": {:hex, :ex_doc, "0.17.1", "39f777415e769992e6732d9589dc5846ea587f01412241f4a774664c746affbb", [:mix], [{:earmark, "~> 1.1", [hex: :earmark, repo: "hexpm", optional: false]}], "hexpm"},
-  "inch_ex": {:hex, :inch_ex, "0.5.5", "b63f57e281467bd3456461525fdbc9e158c8edbe603da6e3e4671befde796a3d", [:mix], [{:poison, "~> 1.5 or ~> 2.0 or ~> 3.0", [hex: :poison, repo: "hexpm", optional: false]}], "hexpm"},
+%{"connection": {:hex, :connection, "1.0.4", "a1cae72211f0eef17705aaededacac3eb30e6625b04a6117c1b2db6ace7d5976", [:mix], []},
+  "db_connection": {:hex, :db_connection, "1.1.2", "2865c2a4bae0714e2213a0ce60a1b12d76a6efba0c51fbda59c9ab8d1accc7a8", [:mix], [{:connection, "~> 1.0.2", [hex: :connection, optional: false]}, {:poolboy, "~> 1.5", [hex: :poolboy, optional: true]}, {:sbroker, "~> 1.0", [hex: :sbroker, optional: true]}]},
+  "decimal": {:hex, :decimal, "1.4.0", "fac965ce71a46aab53d3a6ce45662806bdd708a4a95a65cde8a12eb0124a1333", [:mix], []},
+  "earmark": {:hex, :earmark, "1.2.3", "206eb2e2ac1a794aa5256f3982de7a76bf4579ff91cb28d0e17ea2c9491e46a4", [:mix], []},
+  "ex_const": {:hex, :ex_const, "0.1.0", "7b04d0b65f00cfb8afc0b778bd58dec4804612a54f57122524ee097e9f2c5443", [:mix], []},
+  "ex_doc": {:hex, :ex_doc, "0.17.1", "39f777415e769992e6732d9589dc5846ea587f01412241f4a774664c746affbb", [:mix], [{:earmark, "~> 1.1", [hex: :earmark, optional: false]}]},
+  "ex_spirit": {:hex, :ex_spirit, "0.3.3", "89bd061c68a63b8a4321d768d87e918a7ba60f5a044e5961b14544d1f062901d", [:mix], []},
+  "exconstructor": {:hex, :exconstructor, "1.1.0", "272623a7b203cb2901c20cbb92c5c3ab103cc0087ff7c881979e046043346752", [:mix], []},
+  "html_entities": {:hex, :html_entities, "0.3.0", "2f278ffc69c3f0cbd5996628fef37cf922fa715f3e04b9f38924c8adced3f25a", [:mix], []},
+  "inch_ex": {:hex, :inch_ex, "0.5.5", "b63f57e281467bd3456461525fdbc9e158c8edbe603da6e3e4671befde796a3d", [:mix], [{:poison, "~> 1.5 or ~> 2.0 or ~> 3.0", [hex: :poison, optional: false]}]},
+  "makedown": {:hex, :makedown, "0.2.0", "f47c34e6cc6ddb81a6306c63a580e49fb2c9eea0eec841e579891f1899477280", [:mix], [{:earmark, "~> 1.1", [hex: :earmark, optional: false]}, {:makeup, "~> 0.2.0", [hex: :makeup, optional: false]}]},
+  "makeup": {:hex, :makeup, "0.2.0", "558327c02cd4bc75e54d6aa2e31752c0a906eb4b40b392237921348c4ce455a9", [:mix], [{:ex_const, "~> 0.1.0", [hex: :ex_const, optional: false]}, {:ex_spirit, "~> 0.3.0", [hex: :ex_spirit, optional: false]}, {:exconstructor, "~> 1.1.0", [hex: :exconstructor, optional: false]}, {:html_entities, "~> 0.3.0", [hex: :html_entities, optional: false]}]},
```

Comment:
This project seems to bring a lot of dependencies. 🤔

Reply:
Well, that's what you need if you want a syntax highlighter in pure elixir... I'm sure the Javascript code that highlights it in the browser is equally big, you only don't notice because you don't see them in the mix.exs file.

Reply:
* makedown is an alternative markdown implementation which is basically earmark +  makeup for syntax highlighting.
* makeup is the syntax highighter. It needs html_entities because it has to escape HTML entities inside the code. Yes, I could implement that one myself, if you want to eliminate it as a dependency. inch_ex was already a dependency before. It needs ex_spirit to parse the source code into a list of tokens.
* makeup_elixir is the module responsible for lexing Elixir code (makeup is now modular and not a monolithic package)

Reply:
I don't see how we can reduce the "dependency count" of all of all of this. Highlighting elixir well (the JS implementation is really crappy) takes a lot of code. This code has to actually live somewhere.

The logical step would be to add makedown (the markdown "implementation" that uses makeup) to ExDoc as an "officially supported" markdown implementation. That way, users of ExDoc don't need to import anything else. But this is just sweeping the problem under the rug: it's still a lot of code (from 4 packages and at least 3 different authors), which must be vetted, reviewed, etc. to be included in something as "fundamental" as ExDoc.

You (the ExDoc) team would have the added burden to vet and maintain makedown (i.e. makedown itself, makeup and makeup_elixir) as although I plan on mantaining makedown indefinitely, I can't commit enough time to join the ExDoc team myself as a full-time maintainer.

Given the fact that these dependencies have no impact on anything that happens at runtime, I don't think they're very dangerous, but ultimately it's your call.